### PR TITLE
Change SourceDirectorySet type back to def because this raises an error.

### PR DESCRIPTION
**Pull request by confile**

### CODE REVIEW

**Comment by confile** [comment]
@brunobowden @advayDev1 please have a look.

**Comment by brunobowden** [comment]
What error do you see? We should fix that error rather than go back to `def`.

**Comment by confile** [comment]
```
* What went wrong:
Could not determine the dependencies of task ':j2objcTranslate'.
> Cannot cast object 'file tree' with class 'org.gradle.api.internal.file.UnionFileTree' to class 'org.gradle.api.file.SourceDirectorySet'
```

**Comment by advayDev1** [comment]
Can you run with gradlew with --stacktrace and paste the result?

UnionFileTree and DefaultSourceDirectorySet have common ancestors but this is very weird.  SourceSet's properties are documented as returning SourceDirectorySet.

**Comment by confile** [comment]
@advayDev1 sure here it is: 

```
* What went wrong:
Could not determine the dependencies of task ':j2objcTranslate'.
> Cannot cast object 'file tree' with class 'org.gradle.api.internal.file.UnionFileTree' to class 'org.gradle.api.file.SourceDirectorySet'

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.GradleException: Could not determine the dependencies of task ':j2objcTranslate'.
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:65)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:53)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.realizedDependencies(DefaultTaskExecutionPlan.java:181)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.addToTaskGraph(DefaultTaskExecutionPlan.java:138)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.addTasks(DefaultTaskGraphExecuter.java:76)
	at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.access$100(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:48)
	at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.access$100(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:48)
	at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:44)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.select(DefaultBuildExecuter.java:35)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:142)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:90)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:50)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:27)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:40)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:169)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'file tree' with class 'org.gradle.api.internal.file.UnionFileTree' to class 'org.gradle.api.file.SourceDirectorySet'
	at com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask.getSrcFiles(J2objcTranslateTask.groovy:45)
	at com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask_Decorated.getSrcFiles(Unknown Source)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$PropertyInfo$4.create(AnnotationProcessingTaskFactory.java:460)
	at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:166)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$PropertyInfo.getValue(AnnotationProcessingTaskFactory.java:458)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$Validator$1.call(AnnotationProcessingTaskFactory.java:262)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.doResolve(DefaultFileCollectionResolveContext.java:120)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.resolveAsFileTrees(DefaultFileCollectionResolveContext.java:75)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext$BuildableFileTreeInternalConverter.convertInto(BuildDependenciesOnlyFileCollectionResolveContext.java:50)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.doResolve(DefaultFileCollectionResolveContext.java:98)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.resolveAsFileCollections(DefaultFileCollectionResolveContext.java:82)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.resolveAsBuildables(BuildDependenciesOnlyFileCollectionResolveContext.java:43)
	at org.gradle.api.internal.file.CompositeFileCollection.addDependencies(CompositeFileCollection.java:136)
	at org.gradle.api.internal.file.CompositeFileCollection$3.resolve(CompositeFileCollection.java:124)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:86)
	at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:200)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:112)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:64)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doResolve(CachingTaskDependencyResolveContext.java:74)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:63)
	... 40 more


BUILD FAILED

Total time: 5.258 secs
```

**Comment by brunobowden** [comment]
Here's the error:

Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException:
Cannot cast object 'file tree' with class
'org.gradle.api.internal.file.UnionFileTree' to class
'org.gradle.api.file.SourceDirectorySet'
    at
com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask.getSrcFiles(J2objcTranslateTask.groovy:45)
    at
com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask_Decorated.getSrcFiles(Unknown
Source)
    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)

It's the second line here that's causing trouble:

    SourceDirectorySet allFiles = J2objcUtils.srcDirs(project, 'main',
'java')
    allFiles = allFiles.plus(J2objcUtils.srcDirs(project, 'test', 'java'))

They are related

    SourceDirectorySet extends FileTree

This is the method it's calling:

    FileTree plus(FileTree fileTree);

Does it work if you try this:

    allFiles = allFiles.plus((FileTree) J2objcUtils.srcDirs(project,
'test', 'java'))

**Comment by advayDev1** [comment]
no, don't cast anything.  just change allFiles type to be FileTree.  I think this happens in three files:
https://github.com/j2objc-contrib/j2objc-gradle/search?utf8=%E2%9C%93&q=%22SourceDirectorySet+allFiles%22

**Comment by advayDev1** [comment]
@confile, question: did you build the plugin, and then also test it on your local project before merging https://github.com/j2objc-contrib/j2objc-gradle/pull/206/files#diff-00893f84080bdf45f41d24d7d3e10afaR39 ?  i think we can catch stuff like this if we do that as a practice until our full integration testing is ready.

**Comment by confile** [comment]
@advayDev1 I did test the early changes in the commit but not the last one. Sorry for that.

**Comment by advayDev1** [comment]
please change to FileTree and try that out.  but be sure to change all the instances I pointed out.

**Comment by confile** [comment]
@advayDev1 I tried FileTree but it gets the same error:

**Comment by confile** [comment]
Cannot cast object 'file collection' with class 'org.gradle.api.internal.file.UnionFileCollection' to class 'org.gradle.api.file.FileTree'

**Comment by advayDev1** [comment]
again, full stack trace please.  I assume another one of these needs to be even broader as a FileCollection

**Comment by confile** [comment]
Yes: 

```
Could not determine the dependencies of task ':j2objcTranslate'.
> Cannot cast object 'file collection' with class 'org.gradle.api.internal.file.UnionFileCollection' to class 'org.gradle.api.file.FileTree'

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.GradleException: Could not determine the dependencies of task ':j2objcTranslate'.
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:65)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:53)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.realizedDependencies(DefaultTaskExecutionPlan.java:181)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.addToTaskGraph(DefaultTaskExecutionPlan.java:138)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.addTasks(DefaultTaskGraphExecuter.java:76)
	at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.access$100(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:48)
	at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.access$100(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:48)
	at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:44)
	at org.gradle.execution.DefaultBuildExecuter.configure(DefaultBuildExecuter.java:42)
	at org.gradle.execution.DefaultBuildExecuter.select(DefaultBuildExecuter.java:35)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:142)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:90)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:50)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:27)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:40)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:169)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'file collection' with class 'org.gradle.api.internal.file.UnionFileCollection' to class 'org.gradle.api.file.FileTree'
	at com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask.getSrcFiles(J2objcTranslateTask.groovy:51)
	at com.github.j2objccontrib.j2objcgradle.tasks.J2objcTranslateTask_Decorated.getSrcFiles(Unknown Source)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$PropertyInfo$4.create(AnnotationProcessingTaskFactory.java:460)
	at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:166)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$PropertyInfo.getValue(AnnotationProcessingTaskFactory.java:458)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$Validator$1.call(AnnotationProcessingTaskFactory.java:262)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.doResolve(DefaultFileCollectionResolveContext.java:120)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.resolveAsFileTrees(DefaultFileCollectionResolveContext.java:75)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext$BuildableFileTreeInternalConverter.convertInto(BuildDependenciesOnlyFileCollectionResolveContext.java:50)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.doResolve(DefaultFileCollectionResolveContext.java:98)
	at org.gradle.api.internal.file.collections.DefaultFileCollectionResolveContext.resolveAsFileCollections(DefaultFileCollectionResolveContext.java:82)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.resolveAsBuildables(BuildDependenciesOnlyFileCollectionResolveContext.java:43)
	at org.gradle.api.internal.file.CompositeFileCollection.addDependencies(CompositeFileCollection.java:136)
	at org.gradle.api.internal.file.CompositeFileCollection$3.resolve(CompositeFileCollection.java:124)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:86)
	at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:200)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:112)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:64)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doResolve(CachingTaskDependencyResolveContext.java:74)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:63)
	... 40 more


BUILD FAILED
```

**Comment by advayDev1** [comment]
also @confile , those are NOT the same errors.  earlier you got
Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'file tree' with class 'org.gradle.api.internal.file.UnionFileTree' to class 'org.gradle.api.file.SourceDirectorySet'

not you got
Cannot cast object 'file collection' with class 'org.gradle.api.internal.file.UnionFileCollection' to class 'org.gradle.api.file.FileTree'

One was for UnionFileTree, another for UnionFileCollection

**Comment by confile** [comment]
Sorry I meant I also get an error :-)

**Comment by advayDev1** [comment]
Ah, so just use FileCollection in all the places I mentioned.  The problem is plus(FileCollection) returns a FileCollections, while plus(FileTree) returns a FileTree.  ANyway they all subclass/interface from FileCollection, and even though FileCollection is missing a matching(...) function, since Groovy uses duck typing, we should be fine.

**Comment by advayDev1** [comment]
This is where unit-tests will be critical.  Once we leave the world of 'def' we don't take advantage of Groovy duck-typing as much, and edge-case codepaths will have issues like this.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/J2objcCycleFinderTask.groovy [hunk 1]
```diff
@@ -19,7 +19,7 @@ package com.github.j2objccontrib.j2objcgradle.tasks
 import org.gradle.api.DefaultTask
 import org.gradle.api.InvalidUserDataException
 import org.gradle.api.file.FileCollection
-import org.gradle.api.file.SourceDirectorySet
+//import org.gradle.api.file.SourceDirectorySet
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/J2objcCycleFinderTask.groovy]: please don't leave commented lines in a commit

**Comment by confile** [comment]
I used ``FileCollection``: still getting: 

```
Execution failed for task ':shared:compileIos_arm64DebugShared-j2objcStaticLibraryShared-j2objcObjc'.
> Multiple build operations failed.
      Objective-C compiler failed while compiling SharedClass.m.
      Objective-C compiler failed while compiling Empty.m.
  See the complete log at: file:///Users/mg/Documents/Android/Android-Studio/Sample/shared/build/tmp/compileIos_arm64DebugShared-j2objcStaticLibraryShared-j2objcObjc/output.txt

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':shared:compileIos_arm64DebugShared-j2objcStaticLibraryShared-j2objcObjc'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:42)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.AbstractTask.executeWithoutThrowingTaskFailure(AbstractTask.java:310)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.executeTask(AbstractTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:63)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:51)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:23)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:88)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:62)
	at org.gradle.execution.DefaultBuildExecuter.access$200(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$2.proceed(DefaultBuildExecuter.java:68)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:62)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:55)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:149)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:90)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:50)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:27)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:40)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:169)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
Caused by: org.gradle.internal.operations.MultipleBuildOperationFailures: Multiple build operations failed.
    Objective-C compiler failed while compiling SharedClass.m.
    Objective-C compiler failed while compiling Empty.m.
See the complete log at: file:///Users/mg/Documents/Android/Android-Studio/Sample/shared/build/tmp/compileIos_arm64DebugShared-j2objcStaticLibraryShared-j2objcObjc/output.txt
	at org.gradle.internal.operations.DefaultBuildOperationQueue.waitForCompletion(DefaultBuildOperationQueue.java:72)
	at org.gradle.nativeplatform.toolchain.internal.NativeCompiler.execute(NativeCompiler.java:73)
	at org.gradle.nativeplatform.toolchain.internal.NativeCompiler.execute(NativeCompiler.java:39)
	at org.gradle.nativeplatform.toolchain.internal.OutputCleaningCompiler.compileSources(OutputCleaningCompiler.java:46)
	at org.gradle.nativeplatform.toolchain.internal.OutputCleaningCompiler.execute(OutputCleaningCompiler.java:38)
	at org.gradle.nativeplatform.toolchain.internal.OutputCleaningCompiler.execute(OutputCleaningCompiler.java:26)
	at org.gradle.language.nativeplatform.internal.incremental.IncrementalNativeCompiler.doCleanIncrementalCompile(IncrementalNativeCompiler.java:96)
	at org.gradle.language.nativeplatform.internal.incremental.IncrementalNativeCompiler.execute(IncrementalNativeCompiler.java:74)
	at org.gradle.language.nativeplatform.internal.incremental.IncrementalNativeCompiler.execute(IncrementalNativeCompiler.java:37)
	at org.gradle.nativeplatform.internal.BuildOperationLoggingCompilerDecorator.execute(BuildOperationLoggingCompilerDecorator.java:38)
	at org.gradle.nativeplatform.internal.BuildOperationLoggingCompilerDecorator.execute(BuildOperationLoggingCompilerDecorator.java:22)
	at org.gradle.language.nativeplatform.tasks.AbstractNativeCompileTask.doCompile(AbstractNativeCompileTask.java:115)
	at org.gradle.language.nativeplatform.tasks.AbstractNativeCompileTask.compile(AbstractNativeCompileTask.java:107)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.doExecute(AnnotationProcessingTaskFactory.java:243)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:219)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.execute(AnnotationProcessingTaskFactory.java:230)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:208)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 46 more
Caused by: org.gradle.nativeplatform.toolchain.internal.CommandLineToolInvocationFailure: Objective-C compiler failed while compiling SharedClass.m.
	at org.gradle.nativeplatform.toolchain.internal.DefaultCommandLineToolInvocationWorker.execute(DefaultCommandLineToolInvocationWorker.java:83)
	at org.gradle.nativeplatform.toolchain.internal.DefaultCommandLineToolInvocationWorker.execute(DefaultCommandLineToolInvocationWorker.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$OperationHolder.run(DefaultBuildOperationQueue.java:110)
	at org.gradle.internal.concurrent.ExecutorPolicy$PropagateFailures.onExecute(ExecutorPolicy.java:76)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)


BUILD FAILED


```

**Comment by advayDev1** [comment]
You are going to need to look at:
file:///Users/mg/Documents/Android/Android-Studio/Sample/shared/build/tmp/compileIos_arm64DebugShared-j2objcStaticLibraryShared-j2objcObjc/output.txt

as it says.

Doubt it is related to this though

**Comment by brunobowden** [comment]
@confile - closing this as it's superseded by PR #220
